zaptest/observer/observer_test.go

```diff
@@ -219,6 +227,24 @@ func TestFilters(t *testing.T) {
 			filtered: sink.FilterFieldKey("filterMe"),
 			want:     logs[7:9],
 		},
+		{
+			msg: "filter by arbitrary function",
+			filtered: sink.Filter(func(e LoggedEntry) bool {
+				return len(e.Context) > 1
+			}),
+			want: func() []LoggedEntry {
+				// Do not modify logs slice.
+				w := []LoggedEntry{}
```

Comment:
nit: can use nil slice or a pre-allocated slice here:
```suggestion
				w := make([]LoggedEntry, 0, len(logs))
```

Or,

```suggestion
				var w []LoggedEntry
```